doc/embedded/supported_boards.md

```diff
@@ -0,0 +1,171 @@
+## Setting up an mlpack cross-compilation environment
+
+The following table contains a list of different architectures that you can
+compile or cross compile mlpack to, this page assumes that you have a
+functional operating system (e.g., Linux, MacOS, Windows) with compatible
+ABI on both the host and the target. In addition, your objective is to create
+an embedded ABI (eabi). For now, this guide does not support none-eabi or
+bare-metal C/C++. The following table refers only to possible values of 
+`BOARD_NAME` CMake flag that needs to be specified before the compilation
+process starts.
+
+If your device has a fully functional package manager such as apt (for Debian
+systems), and you have enough RAM on your embedded device (e.g., RAM > 4GB),
+then you will probably be able to compile directly on the device. However, if
+this is not the case, then you will need to follow these instructions for
+cross-compilation.
+
+Setting up the cross compilation toolchain is generally easy, especially when
+using [bootlin toolchains](https://toolchains.bootlin.com/); but other
+toolchains work too.  Once you have the toolchain set up, then you need to
+identify two parameters to configure mlpack with CMake:
+
+  * `TOOLCHAIN_PREFIX`: this specifies the prefix to use when calling compilers and other tools inside the toolchain
+  * `CMAKE_SYSROOT`: this specifies the system root for the cross-compilation environment; in the Bootlin toolchains, this is the `sysroot/` directory
+  
+For more detailed information on these options see [this tutorial](crosscompile_armv7.md).
+
+You can use the table below with your desired architecture to find links to
+appropriate Bootlin toolchains, plus the `TOOLCHAIN_PREFIX` and `CMAKE_SYSROOT`
+options you can use with those toolchains.  If you have provided your own
+toolchain, you will need to adapt the options accordingly.
+
+Once you have found the correct `TOOLCHAIN_PREFIX` and `CMAKE_SYSROOT` options,
+adapt the CMake command below.  ***Don't forget to change
+`/path/to/bootlin/toolchain/` to the correct path on your system!***
+
+```sh
+cmake \
+    -DBUILD_TESTS=ON \
+    -DBOARD_NAME="ARM11" \
+    -DCMAKE_CROSSCOMPILING=ON \
+    -DCMAKE_TOOLCHAIN_FILE=../board/crosscompile-toolchain.cmake \
+    -DTOOLCHAIN_PREFIX=(Check the following table) \
+    -DCMAKE_SYSROOT=(Check the following table) \
+    ../
+```
+
+|-----------------------------------------------------------------------------
+| Architecture\_NAME | link to crosscompiler | CMake command | Example applications |
+|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| ARM11    | [Bootlin toolchain link](https://toolchains.bootlin.com/releases_armv6-eabihf.html) | [Sysroot and toolchain prefix](#ARM11) | [ARM11 on Wikipedia](https://en.wikipedia.org/wiki/ARM11) |
```

Comment:
```suggestion
| ARM11    | [Bootlin toolchain link](https://toolchains.bootlin.com/releases_armv6-eabihf.html) | [Sysroot and toolchain prefix](#arm11) | [ARM11 on Wikipedia](https://en.wikipedia.org/wiki/ARM11) |
```
Ah, the Markdown processor lowercases all of the anchor names, so if you fix them all like this it should fix the documentation build. :+1: